woocommerce/woocommerce-android
# [REST API] Display the Jetpack benefits banner and modal for Application Passwords users

Closes: #8365 

### Description
Internal docs: pe5sF9-19y-p2

As explained in the issue, these are the tasks that this PR does:
- Display the banner when application passwords are used.
- Update the modal's content depending on the site's type.
- Fetch the Jetpack API to detect the status of the Jetpack before attempting the authentication and installation.
- Display the "install Jetpack" button in the settings

The other main change is: adding a feature flag for the project. 

And lastly, although it wasn't a fundamental necessity, I migrated the screen to Compose, this just made handling the different states and the loading dialog easier.

**Note**: the number of lines is misleading, it's just because I moved the files to a different package to group them.

### Testing instructions
1. Open the app, and click on "enter a site address".
2. Enter the address of a non-Jetpack site.
3. Enter your site's credentials.
4. In the dashboard screen, confirm that you can see the Jetpack benefits banner.
5. Click on the banner.
6. Check the content of the modal (the icon for the third point needs updating, I'll update it in a separate PR when we get it from the design team).
7. Click on "Log in to continue"
8. Confirm a loading dialog is shown, then a toast is shown.
9. Go back to the dashboard.
10. Open the settings screen.
11. Confirm that the "Install Jetpack" button is shown.
12. Click on the button, and confirm it opens the Jetpack benefits modal.

### Images/gif
<img width=400 src="https://user-images.githubusercontent.com/1657201/220614856-d0fd2f35-cd5d-4349-9862-e935d9bb5004.png"/>
